lib/cosmos/dart/processes/dart_ingester.rb

```diff
@@ -16,20 +16,57 @@
 # Handles packets by writing them to the dart log file. New SYSTEM META packets
 # cause a new log file to be started.
 class DartInterfaceThread < Cosmos::InterfaceThread
+  include DartCommon
+  
   attr_writer :packet_log_writer
   attr_writer :log_type
-
+  
   def handle_packet(packet)
     if packet.target_name == 'SYSTEM'.freeze and packet.packet_name == 'META'.freeze
       Cosmos::Logger.info("#{@log_type}: #{packet.target_name} #{packet.packet_name}")
+      
+      # Manually read the configuration from the buffer because the packet might not be identified if
+      # identify_and_define is false
+      buffer = packet.buffer(false)
+      if buffer.length >= 33
+        system_config_name = Cosmos::BinaryAccessor.read(8, 256, :STRING, buffer, :BIG_ENDIAN)
+        # Switch to this system_config
+        begin
+          switch_and_get_system_config(system_config_name)
+        rescue Exception => e
+          Cosmos::Logger.error(e.formatted)
+          Cosmos::Logger.error("Could not load system_config: #{system_config_name}")
+        end
+      end      
+      
       # Update Current Value Table Used By Packet Log Writer
       cvt_packet = Cosmos::System.telemetry.update!(packet.target_name, packet.packet_name, packet.buffer)
       cvt_packet.received_time = packet.received_time
       cvt_packet.stored = packet.stored
       cvt_packet.extra = packet.extra
+      
       @packet_log_writer.start
       @packet_log_writer.write(cvt_packet)
     else
+      # Define the packet so that packet_time will work correctly
+      begin
+        packet_data = packet.buffer(false)
+        received_time = packet.received_time
+        stored = packet.stored
+        extra = packet.extra
+        if @log_type == :TLM
+          packet = Cosmos::System.telemetry.packet(packet.target_name, packet.packet_name)
+        else
+          packet = Cosmos::System.commands.packet(packet.target_name, packet.packet_name)
+        end
+        packet.buffer = packet_data
+        packet.received_time = received_time
+        packet.stored = stored
+        packet.extra = extra
+      rescue Exception => error
+        # Could not find a definition for this packet
+        Cosmos::Logger.error "Unknown packet #{target_name} #{packet_name}"
+      end
```

Comment:
handle_packet is passed a packet and then you're asking for the System definition of the packet. Why is this done when we already have a Packet object? Your comment about packet_time isn't telling the whole story for me.

Reply:
We have a packet object that is identified but not defined.   It must be defined (with all the items known) for getting packet_time to not just resolve to received time.   That was the issue raised by ticket #944